enterprise/models.py

```diff
@@ -677,8 +679,11 @@ def notify_enrolled_learners(self, catalog_api_user, course_id, users, admin_enr
         course_path = '/courses/{course_id}/course'.format(course_id=course_id)
         params = {}
         # add tap_hint if there is only one IdP attached with enterprise_customer
-        if self.identity_providers.count() == 1:
+        if self.has_single_idp:
             params = {'tpa_hint': self.identity_providers.first().provider_id}
+
+        elif self.has_multiple_idps and self.default_provider_idp:
```

Comment:
What if there are `multiple idps` but there is no `default provider idp`?

Reply:
This is discussed at the first comment in the PR.
https://github.com/edx/edx-enterprise/pull/1276#discussion_r656038394